pybossa/view/projects.py

```diff
@@ -2925,3 +2925,34 @@ def assign_users(short_name):
 
     flash(msg, 'success')
     return redirect_content_type(url_for('.settings', short_name=project.short_name))
+
+@blueprint.route('/<short_name>/quiz-mode', methods=['GET', 'POST'])
+def quiz_mode(short_name):
+    project = project_repo.get_by_shortname(short_name)
+
+    ensure_authorized_to('read', project)
+    ensure_authorized_to('update', project)
+
+    if request.method == 'POST':
+        form = ProjectQuizForm(request.form)
+        if not form.validate():
+            flash("Please fix the errors", 'message')
+        else:
+            field_names = [
+                'enabled',
+                'questions_per_quiz',
+                'correct_answers_to_pass'
+            ]
+            project.info['quiz'] = { field_name: getattr(form, field_name).data for field_name in field_names }
```

Comment:
Does `form.data` contain additional fields?

Reply:
Are you asking why I specified the `field_ names`? I did that because there is an additional field, `csrf_token`, inherited from `Form`

Reply:
Do you see `csrf_token` in `form.data`? When I print it out I see this:
`{'correct_answers_to_pass': 3, 'questions_per_quiz': 8, 'enabled': True}`

Reply:
You're right. I see the same as you do.